### PR TITLE
Add MCP-over-XDS to bootstrap

### PR DESCRIPTION
**Pull request by costinm**
Istiod now supports returning MCP objects over XDS, and adsc supports reading them. 

This PR adds "xds://" prefixed config sources, using MCP-over-XDS. 

In addition, for consistency and more flexibility it allows K8S to also be defined as a config sources. Before 
it was either MCP or K8S. In order to support 'federation' use cases it is useful to have both the primary 
K8S source as well as XDS sources for objects from other sources. 

With this PR it is now possible to run Istiod with no RBAC privileges ( running on a VM or a regular
namespace with no extra RBAC ) - which may help in some caching, remote istiod and tenant use cases.

It is using plain text XDS - while ADSC supports TLS, it seems far simpler to use the envoy sidecar, as was 
the case in 1.5. We can also do it in the go code - but not a priority for now, since other things from agent 
are needed to run unprivileged. Will describe this in a separate RFC - this PR is just a small piece.

Note this is pretty experimental for now - so 1.8 or 1.9. 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -389,8 +390,23 @@ func (s *Server) WaitUntilCompletion() {
...
 func (s *Server) initKubeClient(args *PilotArgs) error {
-	if hasKubeRegistry(args.RegistryOptions.Registries) {
+	hasK8SConfigStore:= false
+	if args.RegistryOptions.FileDir == "" {
+		// If file dir is set - config controller will just use file.
+		meshConfig := s.environment.Mesh()
+		if meshConfig != nil && len(meshConfig.ConfigSources) > 0 {
+			for _, cs := range meshConfig.ConfigSources {
+				if cs.Address == "k8s://" {
+					hasK8SConfigStore = true
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/server.go:403]: nit: this seems to make more sense to be part of the `hasKubeRegistry` function? Any reason to prefer here

Comment (costinm) [reply pilot/pkg/bootstrap/server.go:403]: It's not technically checking if it has kube registry - this is a kube config source.

Maybe rename hasKubeRegistry to 'needsK8S' ? And also check if CA is enabled.

**Comment by costinm** [comment]
Re. conflict - it's whatever we had with MCP. We'll probably need to adjust a bit - but needs more thinking...
The main use case is additive (local configs + remote), and federation - where we'll need to figure out how to merge / patch.

**Pull request merged**